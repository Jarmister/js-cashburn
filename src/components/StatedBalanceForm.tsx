import React from "react";
import BalanceForm from "./BalanceForm";


type Props = { getValues: () => { date: Date, value: number }[], saveValues: (values: { date: Date, value: number }[]) => void };
type State = { value: number };


class StatedBalanceForm extends React.Component<Props, State> {
    saveToMemory(valueFromBalanceForm: number): void {
        const valuesDatesFromMemory = this.props.getValues();
        const valueFromBalanceFormWithDate = {date: new Date(), value: valueFromBalanceForm};
        const valuesToMemory = [valueFromBalanceFormWithDate, ...valuesDatesFromMemory];
        this.props.saveValues(valuesToMemory);
    };

    constructor(props: Props) {
        super(props);
        this.state = {value: 0};

    };

    render() {
        return (
            <div>
                <BalanceForm callback={(value) => {
                    this.saveToMemory(value);
                }}/>
            </div>

        );
    }
}

export default StatedBalanceForm;

