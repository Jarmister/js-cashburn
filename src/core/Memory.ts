
let savedValues= new Array();
let a;
export function saveValues(values: { date: Date, value: number }[]): void {
  savedValues=values;
}

export function getValues(): { date: Date, value: number }[] {
  

    return savedValues;
}
