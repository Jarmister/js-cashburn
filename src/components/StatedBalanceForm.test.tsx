import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React, {ReactElement} from "react";
import StatedBalanceForm from "./StatedBalanceForm";
import {saveValues} from "../core/Memory";


Enzyme.configure({adapter: new Adapter()});

describe('StatedBalanceForm', () => {
    it('renders without crashing given the required props', () => {
        const save_value = jest.fn();
        const get_value = jest.fn();
        const wrapper = shallow(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
        expect((wrapper)).toMatchSnapshot();
    })
})

test("When user inputs, then get_values is called ", () => {
    const save_value = jest.fn();
    const get_value = jest.fn(()=>[]);
    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
    StatedbalanceForm.find("button").simulate("click");
    expect(get_value).toBeCalled();
})

test("When user inputs, then save_value is called", () => {
    const save_value = jest.fn();
    const get_value = jest.fn(()=>[]);
    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
    StatedbalanceForm.find("button").simulate("click");
    expect(save_value).toBeCalled();
})

test("SaveValues save values to Memory's empty storage ", () => {
    const save_value = jest.fn();
    const get_value = jest.fn(()=>[]);
    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
    StatedbalanceForm.find("input").simulate("change", {target: {value: 21}});
    StatedbalanceForm.find("button").simulate("click");
    expect(save_value).toBeCalledWith([{"date": new Date(), "value": 21}]);
})

test("SaveValues save values to Memory's previous values ", () => {
    const save_value = jest.fn();
    const get_value = jest.fn(()=>[{"date": 12-21-2020, "value": 4}]);
    const StatedbalanceForm = mount(<StatedBalanceForm saveValues={save_value} getValues={get_value}/>);
    StatedbalanceForm.find("input").simulate("change", {target: {value: 21}});
    StatedbalanceForm.find("button").simulate("click");
    expect(save_value).toBeCalledWith([{"date": new Date(), "value": 21}, {"date": 12-21-2020, "value": 4}]);
})