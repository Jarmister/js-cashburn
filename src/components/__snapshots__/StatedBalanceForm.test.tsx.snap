// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatedBalanceForm renders without crashing given the required props 1`] = `ShallowWrapper {}`;
